### DISCUSSION TITLE
Documentation and Examples

### DISCUSSION THREAD

**Discussion by bburas** [ni/easyrdma#17] [General]
I'm browsing through the tests, but I wanted to ask if there is any documentation or examples (client/server) etc?

**Comment by cgraf78** [comment]
I created a basic client/server example, which I've been meaning to contribute back to this project.  It still needs a bit of cleanup before I'd call it ready for official review, but I can dig it up if that'd be helpful to you?

**Comment by cgraf78** [comment]
Here are the examples I mentioned:

https://github.com/cgraf78/easyrdma/commit/b1cf019ee24b4450636f38c5eee04a173b007fa0

They're currently sitting in my fork of the repo, but I intend to create a PR to submit them back upstream.  It's been a month or two since I last compiled & used them, but they were fully functional last time I did.

Hope this helps get you up and running!

**Reply by bburas** [reply]
Very helpful. Thanks!

**Comment by bburas** [comment]
Do you have an rpc example of send/reply where both sides perform a send and receive /  receive and send ? Do I need to use 2 sessions with separate buffers?

**Reply by cgraf78** [reply]
If using the send/recv examples posted above, easiest thing to do would be to run two separate instances of each, in opposite directions, each using different ports.

**Comment by nikhalim** [comment]
@cgraf78 Thanks for helping to chime in while I was away!
